src/ranged.cpp

```diff
@@ -918,6 +930,12 @@ int Character::fire_gun( const tripoint &target, int shots, item &gun )
     int hits = 0; // total shots on target
     int delay = 0; // delayed recoil that has yet to be applied
     while( curshot != shots ) {
+        if( !!ammo && !gun.ammo_remaining() ) {
+            Character &you = get_avatar();
+            gun.reload( you, ammo, 1 );
+            you.burn_energy_arms( - gun.get_min_str() * static_cast<int>( 0.006f *
+                                  get_option<int>( "PLAYER_MAX_STAMINA_BASE" ) ) );
+        }
```

Comment:
```suggestion
        // Special handling for weapons where we supply the ammo separately (i.e. ammo is populated)
        // instead of it being loaded into the weapon, reload right before firing.
        if( !!ammo && !gun.ammo_remaining() ) {
            Character &you = get_avatar();
            gun.reload( you, ammo, 1 );
            you.burn_energy_arms( - gun.get_min_str() * static_cast<int>( 0.006f *
                                  get_option<int>( "PLAYER_MAX_STAMINA_BASE" ) ) );
        }
```